### PR TITLE
Skip and publish performance

### PR DESCRIPTION
**Pull request by robinsio**

### CODE REVIEW

File: test/hutch.test.js [hunk 1]
```diff
@@ -280,4 +379,67 @@ describe('Hutch', function() {
...
+        });
+
+        // Send a message to consumer 1 then close it.
+        instance1.publish(options, "Example Message!", function(err, res){
+          instance1.close(options.queue.name, function(err){});
+        });
+
+        setTimeout(function(){
+          // We should have only processed one of the files.
+          messages[0].should.equal('c2');
```
Comment (pa-colin-mchale) [inline test/hutch.test.js:441]: This checks that the first message processed was by consumer 2, not that only one message was processed.

File: test/hutch.test.js [hunk 2]
```diff
@@ -280,4 +379,67 @@ describe('Hutch', function() {
       });
     });
   });
+
+  it('should clear skip option for exclusive consumer after consumer is rejected', function(complete) {
+
+    this.timeout(20000); // Allow up to 15 seconds for windows
```
Comment (pa-colin-mchale) [inline test/hutch.test.js]: Comment has wrong time.

File: test/hutch.test.js [hunk 3]
```diff
@@ -83,6 +83,55 @@ describe('Hutch', function() {
...
+      };
+
+      var count = 0;
+      var consumer = function(message, done, fail) {
+        count++;
+        done();
+
+        if(count === 2){
+          var channel = hutch._getChannelByExchange(options.exchange.name);
+          channel.should.exist();
```
Comment (pa-colin-mchale) [inline test/hutch.test.js]: How does this check it uses the same channel?

Comment (robinsio) [reply test/hutch.test.js]: It doesn't, but it does check to see if we can pull out the channel by exchange which is what happens prior to the publish call

Comment (pa-colin-mchale) [reply test/hutch.test.js]: Maybe change the test description then.

**Pull request merged**